Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -2183,16 +2374,78 @@ private static void ReportFileAccesses(ExecutionResult processExecutionResult, F
                     environment.State.ExecutionLog?.PipCacheMiss(pipCacheMiss.Value);
                 }
 
-                using (operationContext.StartOperation(PipExecutorCounter.CheckProcessRunnableFromCacheExecutionLogDuration))
+                processRunnable.CacheLookupPerfInfo.LogCounters(pipCacheMiss.Value.CacheMissType, numPathSetsDownloaded, numCacheEntriesVisited);
+                return runnableFromCacheResult;
+            }
+        }
+
+        /// <summary>
+        /// Extract a path set to represent the commonly accessed paths which can be used to compute an augmented weak fingerprint
+        /// </summary>
+        private static ObservedPathSet ExtractPathSetForAugmentingWeakFingerprint(
+            PathTable pathTable,
+            ICacheConfiguration cacheConfiguration,
+            List<BoxRef<ProcessStrongFingerprintComputationData>> strongFingerprintComputationList)
+        {
+            var requiredUseCount = Math.Max(1, cacheConfiguration.AugmentWeakFingerprintPathSetThreshold / 2);
+            using (var pool = s_pathToObservationEntryMapPool.GetInstance())
+            using (var stringIdPool = Pools.StringIdSetPool.GetInstance())
+            {
+                Dictionary<AbsolutePath, ExtractedPathEntry> map = pool.Instance;
+                var accessedFileNameSet = stringIdPool.Instance;
+
+                foreach (var pathSet in strongFingerprintComputationList.Select(s => s.Value.PathSet))
                 {
-                    environment.State.ExecutionLog?.ProcessFingerprintComputation(processFingerprintComputationResult);
+                    foreach (var accessedFileName in pathSet.ObservedAccessedFileNames)
+                    {
+                        accessedFileNameSet.Add(accessedFileName);

```

Comment:
>accessedFileNameSet [](start = 24, length = 19)

why do we union all ObservedAccessedFileNames ? the set might contain entries that have no corresponding entry in a newly contructed path set